### PR TITLE
[Decision] Add document for Elixir bindings

### PR DESCRIPTION
**Pull request by 0x6178656c**
## Basics

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [x] The PR is rebased with current master.

## Checklist

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation (see [Documentation Guidelines](https://www.libelektra.org/devgettingstarted/documentation))
- [ ] I fixed all affected decisions (see [Decision Process](https://www.libelektra.org/decisions/decision-process))
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://www.libelektra.org/devgettingstarted/coding))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned [every code](/.reuse/dep5) not directly written by me in [reuse syntax](https://reuse.software/)

## Review

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

- [ ] Add the "work in progress" label if you do not want the PR to be reviewed yet.
- [ ] Add the "ready to merge" label **if the basics are fulfilled** and no further pushes are planned by you.

### CODE REVIEW

File: doc/decisions/0_drafts/elixir_bindings.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+## Decision
+
+The chosen approach was to use "Alternative A", i.e., to implement the NIFs by hand.
+
+## Rationale
+
+The main reason for this choice were time constraints.
+
+Since I have assumed that the availability of the Elixir bindings is not very important, it is acceptable if future changes break these bindings.
```
Comment (markus2330) [inline doc/decisions/0_drafts/elixir_bindings.md:54]: See above, meaning "availability" needs to be clarified. (Here it reads like compatibility.)

File: doc/decisions/0_drafts/elixir_bindings.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+## Implications
+
+- Any change in the C API breaks the Elixir binding which needs to be addressed manually.
```
Comment (markus2330) [inline doc/decisions/0_drafts/elixir_bindings.md:58]:
```suggestion
- Every incompatible change in the C API needs manual changes in the Elixir binding.
```

**Review by markus2330 — CHANGES_REQUESTED** [review]
Very well written decision. Imho your decision was the right one.

You can directly target it to 3_decided, as it is already implemented and hopefully gets merged soon.

**Comment by markus2330** [comment]
There is now an invalid link from the release notes.

**Comment by 0x6178656c** [comment]
> There is now an invalid link from the release notes.

I think this is unrelated to my changes.

```
Broken Links:
—————————————
/tmp/cirrus-ci-build/doc/news/_preparation_next_release.md:496:0 warning: invalid link '../decisions/0_drafts/elixir_bindings.md'
doc/news/2019-02-26_0.8.26.md:296:0 https://www.gnu.org/software/gettext
doc/decisions/0_drafts/man_pages.md:34:0 https://www.gnu.org/software/help2man
doc/news/2018-11-18_0.8.25.md:128:0 https://www.gnu.org/software/bison
doc/news/2019-08-06_0.9.0.md:740:0 https://www.gnu.org/software/bison
src/plugins/README.md:93:0 https://www.gnu.org/software/bison
doc/GETSTARTED.md:22:0 https://www.gnu.org/software/make
```

**Comment by 0x6178656c** [comment]
You are right, I forgot to update a link. I was confused by the output of the link checker.

**Comment by markus2330** [comment]
@flo91 please review (and merge)

File: doc/decisions/3_decided/elixir_bindings.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+## Problem
+
+When creating Elixir bindings one needs a way to use some kind of foreign function interface.
+In Erlang (which is the basis for Elixir) such a foreign function interface exists in the form of NIFs (native implemented functions).
+The question is how this NIF interface should be implemented.
+
+## Constraints
+
+1. Elektra's C API is given.
+2. The implementation should be completed in a given time frame.
```
Comment (flo91) [inline doc/decisions/3_decided/elixir_bindings.md:12]: Very general statement, maybe some interval (e.g. "between 20 and 50 person hours") would give a better indication of how much work is expected.

**Review by flo91 — COMMENTED** [review]
Well written elaboration!

File: doc/decisions/3_decided/elixir_bindings.md (hunk 3 above)
Comment (0x6178656c) [reply doc/decisions/3_decided/elixir_bindings.md]: Thanks. I made a change along those lines.

**Review by markus2330 — APPROVED** [review]
LGTM

**Comment by markus2330** [comment]
jenkins build libelektra please

**Comment by markus2330** [comment]
jenkins build libelektra please

**Pull request merged**